Added new DoD topics #9282 (#9)
AdobeDocs/commerce-contributor: src/pages/guides/code-contributions/definition-of-done.md

---
title: Definition of Done | Commerce Contributor
description: Ensure that your contributions meet the criteria of the Commerce definition of done.
---

> ## Overview

Reviewer: ```suggestion
# Definition of done
```

-  TCO impact should be defined for all changes.

The following sections provide additional details about each of these criteria:

## Review

A task, when complete by the author, should be reviewed by a maintainer through a formal code review.

The reviewer should check whether the task meets the original acceptance criteria, verify that there are no code defects and that all points of this DoD are met.

## Tests

To meet the DoD, all changes to Adobe Commerce projects should be covered by automated tests.

Details on different types of automated tests can be found in [DoD - Automated Tests page][3].

## Documentation

All changes, additions, and extensions to the product should be documented by the author.
The documentation should provide an overview of the change, and information about standard use cases, audience, and procedural instructions for implementing the feature.

Documentation for the submitted code should be submitted as a pull request to the official Magento DevDocs [repository][6].

When submitting either code or documentation, a brief summary of the work should be included in the commit message.

> Additionally, the code itself should follow our [DocBlock standard][1] and contain high-quality comments and descriptions for all classes and methods.

Reviewer: ```suggestion
Additionally, the code itself should follow our [DocBlock standard](https://devdocs.magento.com/guides/v2.4/coding-standards/docblock-standard-general.html) and contain high-quality comments and descriptions for all classes and methods.
```

> For documenting backward-incompatible changes, please see [Backwards Compatible Development][2].

Reviewer: ```suggestion
For documenting backward-incompatible changes, see the [Backward compatibility policy](backward-compatibility-policy.md).
```

## Backward Compatibility

All backward incompatible changes should be approved and covered by static tests.

The details on backward compatibility criteria included in the DoD can be found in [DoD - Backward Compatibility][0]

## TCO Analysis

All functional changes should analyzed for the impact on merchant total cost of ownership (TCO) and the results of the analysis should be added to the issue or pull request description to meet the DoD.

The details on TCO analysis required to meet the DoD can be found in [DoD - TCO Analysis][4]

The work cannot be considered as complete unless all the criteria are verified.

> [0]:./dod-backward-compat.md

Reviewer: ```suggestion
```

> [1]:https://devdocs.magento.com/guides/v2.4/coding-standards/docblock-standard-general.html

Reviewer: ```suggestion
```

> [2]:./backward-compatibility-policy.md

Reviewer: ```suggestion
```

> [3]:./automated-tests.md

Reviewer: ```suggestion
```

[4]:./tco-analysis.md

> [5]:https://github.com/magento/magento-coding-standard

Reviewer: ```suggestion
```

> [6]:https://github.com/magento/devdocs

Reviewer: ```suggestion
```
